### PR TITLE
Feature/drop raw data carbage collection

### PR DESCRIPTION
**Pull request by savpek**
History:
- Originally there were no "raw html" page support with .html ending at results. Which were added to give preview for mobile devices etc use cases.
- It basically made obsolete need for temporary html binary table in sql server since it's basically same thing.

Reason to change:
- To optimize generator speed removing binary data were left for garbage collector, however it introduced problem by itself because it is very expensive to execute and can be problematic because of spikes it cause to workload.

Changed:
- Removed carbage collection, removed temporary binary table and use .html version of file stored (which is available with .html extension at side of PDF).

### CODE REVIEW

File: Pdf/PdfController.cs [hunk 1]
```diff
@@ -85,6 +86,13 @@ public ActionResult<IEnumerable<NewPdfResponse>> AddNewPdf([Required][FromRoute]
             return StatusCode(202, responses.ToList());
```
Comment (Hekku2) [inline Pdf/PdfController.cs]: I would prefer moving 202 this to constant.

Also, couple of typos (`attleast`, `Group id is is`).

File: Pdf/PdfController.cs [hunk 2]
```diff
@@ -147,7 +155,9 @@ public IActionResult Get([FromRoute] string groupId, [FromRoute] string pdfId, [
         [HttpHead("/v1/pdf/{groupId}/{pdfId}.{extension}")]
         public IActionResult GetPdfHead([FromRoute] string groupId, [FromRoute] string pdfId)
         {
-            var pdfEntity = _context.PdfFiles.SingleOrDefault(x => x.GroupId == groupId && x.FileId == pdfId);
+            var pdfEntity = _context.PdfFiles
+                .Select(x => new { x.GroupId, x.FileId, x.Processed })
+                .SingleOrDefault(x => x.GroupId == groupId && x.FileId == pdfId);
```
Comment (Hekku2) [inline Pdf/PdfController.cs]: I assumen there can be multiple `PdfFiles` with same `GroupId` and `FileId`. In this case, if processing the first item fails, this always returns false. Is this by design?

File: Pdf/PdfQueue.cs [hunk 3]
```diff
@@ -17,37 +17,30 @@ public class PdfQueue : IPdfQueue
...
         public void CreatePdf(Guid pdfEntityId)
         {
             var entity = _context.PdfFiles.Single(x => x.Id == pdfEntityId);
-            var rawData = _context.RawData.Single(x => x.ParentId == pdfEntityId);
-
-            var templatedHtml = _templatingEngine.Render(rawData.Html, rawData.TemplateData);
-            templatedHtml = TemplateUtils.AddWaitForAllPageElementsFixToHtml(templatedHtml);
-            var data = GeneratePdfDataFromHtml(pdfEntityId, templatedHtml).GetAwaiter().GetResult();
+            var htmlFromStorage = _storage.Get(new StorageFileId(entity, "html"));
+            var data = GeneratePdfDataFromHtml(pdfEntityId, Encoding.UTF8.GetString(htmlFromStorage.Data)).GetAwaiter().GetResult();
```
Comment (Hekku2) [inline Pdf/PdfQueue.cs]: I assume there is some reason for not converting `CreatePdf` to `async`, related to Hangfire?

File: Pdf/PdfController.cs (hunk 2 above)
Comment (savpek) [reply Pdf/PdfController.cs]: File id is always unique

File: Pdf/PdfQueue.cs (hunk 3 above)
Comment (savpek) [reply Pdf/PdfQueue.cs]: Yeah hangfire async support in 1.x is bit partial.

File: Pdf/PdfController.cs (hunk 2 above)
Comment (Hekku2) [reply Pdf/PdfController.cs]: Okay, then I think we can just `.Any`?

Comment (savpek) [reply Pdf/PdfController.cs]: Theres another check under it `if (!pdfEntity.Processed)` that also results not found (pdf isn't processed yet and not available)

Comment (Hekku2) [reply Pdf/PdfController.cs]: `.Any(x => x.GroupId == groupId && x.FileId == pdfId && x.Processed);`

And both of those return `NotFound()`

File: Pdf/PdfController.cs (hunk 1 above)
Comment (savpek) [reply Pdf/PdfController.cs]: Fixed with `Accepted(responses.ToList())`, dunno why i didn't use that originally 😄

File: Pdf/PdfController.cs (hunk 2 above)
Comment (savpek) [reply Pdf/PdfController.cs]: Truedat 👍

**Pull request merged**